pd/src/components/validator_set.rs

```diff
@@ -246,21 +243,6 @@ impl BlockChanges {
             .await?;
         }
 
-        // TODO: should this be part of a future shielded pool component?
-        // Save any new assets found in the block to the asset registry.
-        for (id, asset) in &self.genesis_supply_updates {
-            query!(
-                "INSERT INTO assets (asset_id, denom, total_supply)
-            VALUES ($1, $2, $3)
-            ON CONFLICT (asset_id) DO UPDATE SET denom=$2, total_supply=$3",
-                &id.to_bytes()[..],
-                asset.0.to_string(),
-                asset.1 as i64
-            )
-            .execute(&mut *dbtx)
-            .await?;
-        }
-
```

Comment:
I didn't see where this logic moved to, so maybe we're not computing the asset balances correctly any more (?), but probably it's easier to fix that as part of the state refactoring, so we should probably merge this in as-is?